### DISCUSSION TITLE
Provider Methods for rust exist?

### DISCUSSION THREAD

**Discussion by GrafMine** [phantom/docs#194] [General]
Hello everyone, can you tell me a couple of things about this code?
https://gist.github.com/GrafMine/cc6ff262660f8143b9915607428c5df9

1.how can I cast the phantom variable into my rust type?
2.Maybe someone has already written their own typification for phantom methods? like rust sdk?

p.s. in my case I'm using Leptos for a client, and I want to use methods of interacting with the phantom wallet

**Comment by adamdelphantom** [comment]
Hello @GrafMine,

We don't have any specific support for a rust client. I would recommend searching online to see if you can locate any community built rust SDKs that ingest the Phantom provider.
